src/Build/Evaluation/LazyItemEvaluator.LazyItemOperation.cs

```diff
@@ -110,14 +110,14 @@ protected void DecorateItemsWithMetadata(ICollection<I> items, ImmutableList<Par
 
                     // Do not expand properties as they have been already expanded by the lazy evaluator upon item operation construction.
                     // Prior to lazy evaluation ExpanderOptions.ExpandAll was used.
-                    const ExpanderOptions metadataExpansionOptions = ExpanderOptions.ExpandMetadata | ExpanderOptions.ExpandItems;
```

Comment:
I have no real idea about this code :( But my cursory glance (and assumptions!) suggested that the property gets expanded at evaluation time, but since metadata is not being expanded at that time, the property just evaluates to the string `%(RelativeDir)`, which would mean that the whatever the property function wanted to do with it, cannot be done now. Isn't that still an issue?

You are correct that it says `ExpandMetadata` here, so even if it expanded to the `string` `%(RelativeDir)`, that should have expanded at this time.

I'll wait for the real fix :) 👍